MODULES-2938/restore target feature

oshaughnessy opened the pull request:
Restore the behavior of the old undocumented `target` parameter.

Change config files to support a `config_file` parameter, which offers the same behavior as the old `target` parameter. `target` was previously available (in at least version 1.0.0), but undocumented, as a way to change the destination for config text.

See https://tickets.puppetlabs.com/browse/MODULES-2938 for reference.

jonnytpuppet commented:
I have a slight issue with the README description. Am I correct in saying that the file must already exist with its desired contents? Or is this purly a file naming issue (i.e. the templates are still used and the properties work as desired)

Thanks

oshaughnessy commented:
Hmm. I just copied a portion of the text that was already there for `haproxy::instance`. I thought it should be consistent.

Looking more closely, though, I don't think the file needs to exist already. In my use case, it is generated completely by the module. I took the text in `haproxy::instance` to mean that the directory component of the given path must already exist; the module won't do anything to create the leading directories.

jonnytpuppet commented:
@oshaughnessy This needs a rebase before it can be merged, please. Also, do you propose that we change the readme to be more clear? Thanks

oshaughnessy commented:
Thanks, Jonny. I'll rebase it and submit it shortly. I'll take a look at the README contents, too. Sounds like that needs to rewording.

HelenCampbell commented:
Hey @oshaughnessy , thank you for your contribution! Since it's been a while since there has been any activity on this PR I thought I'd give you a quick ping. Would you still be willing to rebase and update the README as @jonnytpuppet suggested?

oshaughnessy commented:
Yup. I'm working on it today.

oshaughnessy commented:
I've never done a rebase. I re-pulled from upstream and re-pushed back to my branch. Should I create a new pull request for that?

jonnytpuppet commented:
Hi @oshaughnessy 

This needs re-based and travis to pass before we can merge.

git rebase master will bring in the changes from master in to your own branch. Then you can git push --force to this PR

Thanks

JT

oshaughnessy commented:
Trying again. I'll keep an eye on the travis-ci results and correct as needed.

HelenCampbell commented:
Hey @oshaughnessy , Thank you for all the hard work you are putting in to this, it's really appreciated. I'm offline until Tuesday, but I will gladly give you advice and direction on our git workflow then if you still require it. Have a good weekend!

oshaughnessy commented:
Well, I'm learnin stuff. Debugging goes so much faster by running rspec locally instead of waiting on travis-ci for everything! I added some better inline documentation, rewrote what was added to README, and added a check for `validate_absolute_path()` everywhere `config_file` is given as a parameter.

DavidS commented:
@HelenCampbell if you could have a look at this?

HelenCampbell commented:
Hey @oshaughnessy . Thank you for your patience with this, I have merged a PR to fix the current issues we are seeing within the failing Travis run to do with Parallel test Gems. Since that is the case and Travis is passing on everything bar Ruby 1.9.3 I am gunna go ahead and merge this. Thank you! and hope you enjoyed Puppet Conf!

oshaughnessy commented:
Awesome! This is a proud moment. Puppet Conf was amazing. Are you hiring? :^)

HelenCampbell commented:
Hey @oshaughnessy , absolutely we are! Have a look here: https://puppet.com/company/careers/jobs , see if there's something that piques your interest :)
